### PR TITLE
fix(seed): Typos in testing code

### LINKED ISSUE #16314: React Conditional Rendering from Props
#### Challenge Name
React: Render Conditionally from Props

#### Issue Description

It seems code working but there is a problem in the [6th test, assert part](https://github.com/freeCodeCamp/freeCodeCamp/blob/16dcfb41833498e0a6efd788735b413900970817/seed/challenges/03-front-end-libraries/react.json#L2618)
` 'Turn: N' ` should be ` \'Turn: N\'`

#### Browser Information

* Browser Name, Version: Opera 49.0.2725.64
* Operating System: Ubuntu 16.04
* Mobile, Desktop, or Tablet: Desktop

#### Your Code
```js
class Results extends React.Component {
  constructor(props) {
    super(props);
  }
  render() {
    const { fiftyFifty } = this.props;
    return (
      <h1>
      {
        fiftyFifty ? 'You Win!' : 'You Lose!'
      }
      </h1>
    )
  };
};

class GameOfChance extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      counter: 1
    }
    this.handleClick = this.handleClick.bind(this);
  }
  handleClick() {
    this.setState({
      counter: this.state.counter + 1
    });
  }
  render() {
    let expression = Math.random() > .5;
    return (
      <div>
        <button onClick={this.handleClick}>Play Again</button>
        { /* change code below this line */ }
        <Results fiftyFifty={expression} />
        { /* change code above this line */ }
        <p>{'Turn: ' + this.state.counter}</p>
      </div>
    );
  }
};
```
#### Screenshot

image

### LINKED ISSUE #16316: React Filter an Array
#### Challenge Name

[React: Use Array.filter() to Dynamically Filter an Array](https://beta.freecodecamp.org/en/challenges/react/use-arrayfilter-to-dynamically-filter-an-array)

#### Issue Description
Similar syntax error with #16314 and #16312 in [testing](https://github.com/freeCodeCamp/freeCodeCamp/blob/16dcfb41833498e0a6efd788735b413900970817/seed/challenges/03-front-end-libraries/react.json#L2901)

#### Browser Information

* Browser Name, Version: Opera 49.0.2725.64
* Operating System: Ubuntu 16.04
* Mobile, Desktop, or Tablet: Desktop

#### Your Code

```js
class MyComponent extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      users: [
        {
          username: 'Jeff',
          online: true
        },
        {
          username: 'Alan',
          online: false
        },
        {
          username: 'Mary',
          online: true
        },
        {
          username: 'Jim',
          online: false
        },
        {
          username: 'Sara',
          online: true
        },
        {
          username: 'Laura',
          online: true
        }
      ]
    }
  }
  render() {
    const usersOnline = this.state.users.filter(u => u.online);
    const renderOnline = usersOnline.map(u => <li key={u.username}>{u.username}</li>);
    return (
       <div>
         <h1>Current Online Users:</h1>
         <ul>
           {renderOnline}
         </ul>
       </div>
    );
  }
};
```
#### Screenshot

image

### PR DESCRIPTION
**Pull request by raisedadead**
This is a commit, made by @r1chard5mith I wanted to ensure, the due credit goes to them for the efforts, and they do not have to work on this again.

Closes #16314
Closes #16316

#### Pre-Submission Checklist

- [ ] Your pull request targets the `staging` branch of freeCodeCamp.
- [ ] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [ ] You have only one commit (if not, [squash](http://forum.freecodecamp.org/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [ ] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [ ] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [ ] Tested changes locally.
- [ ] Addressed currently open issue (replace XXXXX with an issue no in next line)

Closes #XXXXX

### CODE REVIEW

File: seed/challenges/03-front-end-libraries/react.json [hunk 1]
```diff
@@ -2615,7 +2615,7 @@
...
..., 250)); const comp = Enzyme.mount(React.createElement(GameOfChance)); const simulate = () => { comp.find('button').simulate('click'); };const result = () => ({ count: comp.state('counter'), text: comp.find('p').text() });const _1 = () => { simulate(); return waitForIt(() => result())}; const _2 = () => { simulate(); return waitForIt(() => result())}; const _3 = () => { simulate(); return waitForIt(() => result())}; const _4 = () => { simulate(); return waitForIt(() => result())}; const _5 = () => { simulate(); return waitForIt(() => result())}; const _1_val = await _1(); const _2_val = await _2(); const _3_val = await _3(); const _4_val = await _4(); const _5_val = await _5(); assert(_1_val.count === 2 && _1_val.text === 'Turn: 2' && _2_val.count === 3 && _2_val.text === 'Turn: 3' && _3_val.count === 4 && _3_val.text === 'Turn: 4' && _4_val.count === 5 && _4_val.text === 'Turn: 5' && _5_val.count === 6 && _5_val.text === 'Turn: 6', 'message: Each time the button is clicked, the counter state should be incremented by a value of 1, and a single <code>p</code> element should be rendered to the DOM that contains the text \"Turn: N\", where N is the value of the counter state.'); }; ",
```
Comment (no-stack-dub-sack) [inline seed/challenges/03-front-end-libraries/react.json]: Wow! Nice catch! Even with the diff I could barely see it.

File: seed/challenges/03-front-end-libraries/react.json [hunk 2]
```diff
@@ -2333,9 +2333,9 @@
...
+        "assert.strictEqual(Enzyme.mount(React.createElement(MagicEightBall)).children().childAt(0).name(), 'input', 'message: <code>MagicEightBall</code>&apos;s first child should be an <code>input</code> element.');",
+        "assert.strictEqual(Enzyme.mount(React.createElement(MagicEightBall)).children().childAt(2).name(), 'button', 'message: <code>MagicEightBall</code>&apos;s third child should be a <code>button</code> element.');",
+        "assert(Enzyme.mount(React.createElement(MagicEightBall)).state('randomIndex') === '' && Enzyme.mount(React.createElement(MagicEightBall)).state('userInput') === '', 'message: <code>MagicEightBall</code>&apos;s state should be initialized with a property of <code>userInput</code> and a property of <code>randomIndex</code> both set to a value of an empty string.');",
```
Comment (no-stack-dub-sack) [inline seed/challenges/03-front-end-libraries/react.json]: This works - I had them escaped before I started using the html entity for the apostrophe

File: seed/challenges/03-front-end-libraries/react.json [hunk 3]
```diff
@@ -2898,7 +2898,7 @@
...
+        "async () => { const waitForIt = (fn) => new Promise((resolve, reject) => setTimeout(() => resolve(fn()), 250)); const comp = Enzyme.mount(React.createElement(MyComponent)); const users = (bool) => ({users:[ { username: 'Jeff', online: bool }, { username: 'Alan', online: bool }, { username: 'Mary', online: bool }, { username: 'Jim', online: bool   }, { username: 'Laura', online: bool } ]}); const ul = () => { comp.setState(users(true)); return waitForIt(() => comp.find('ul').html()) }; const html = await ul(); assert(html === '<ul><li>Jeff</li><li>Alan</li><li>Mary</li><li>Jim</li><li>Laura</li></ul>', 'message: <code>MyComponent</code> should render <code>li</code> elements that contain the username of each online user.'); }; ",
```
Comment (no-stack-dub-sack) [inline seed/challenges/03-front-end-libraries/react.json]: Also a nice catch - strange, because I thought this challenge was working!

**Comment by no-stack-dub-sack** [comment]
LGTM! Thanks @raisedadead & @r1chard5mith!

Thanks @SanD94 for catching the bug

**Comment by no-stack-dub-sack** [comment]
@raisedadead @r1chard5mith I just caught up on the original issue thread, if I was premature in merging this, please let me know. We can always revert and make additional changes if need be.

**Comment by raisedadead** [comment]
@no-stack-dub-sack lets take more changes in a fresh PR, this was created to preserve the original work of @r1chard5mith

**Comment by raisedadead** [comment]
@no-stack-dub-sack thanks a lot for the QA and the merge.

**Comment by no-stack-dub-sack** [comment]
@raisedadead no worries - thanks!

**Pull request merged**